### PR TITLE
Reorganize Python SDK documentation, add more details

### PR DESCRIPTION
**Pull request by johndmulhausen**
## Summary

This PR reorganizes the Python SDK documentation structure to improve clarity and navigation, added usage details and API positioning/comparison 

### Impact:
- Improved documentation organization with clearer hierarchy
- Better grouping of related SDK documentation
- All existing links preserved through redirects

### Testing:
- All documentation builds successfully
- Internal links have been updated and verified

Fixes documentation structure issues identified in the new reference table of contents project.

📄 **[View preview links for changed pages](https://github.com/wandb/docs/pull/1611#issuecomment-3269042718)**

### CODE REVIEW

File: content/en/ref/python/wandb_workspaces/_index.md [hunk 1]
```diff
@@ -1,17 +1,47 @@
...
+```python
+import wandb_workspaces
+```
+
+Specify the entity of your team in the following code block:
+
+```python
+TEAM_ENTITY = "<Team_Entity>" # Replace with your team entity
+PROJECT = "my-awesome-project"
+```
```
Comment (ngrayluna) [inline content/en/ref/python/wandb_workspaces/_index.md]: This is how we currently show users how to import workspace and reports API

```python
import wandb_workspaces.workspaces as ws
import wandb_workspaces.reports.v2 as wr 
```

File: content/en/ref/python/wandb_workspaces/_index.md [hunk 2]
```diff
@@ -1,17 +1,47 @@
...
+### Sign up and create an API key
+
+To authenticate your machine with W&B, you must first generate an API key at https://wandb.ai/authorize.
+
+### Install and import packages
+
+Install the W&B library.
+
+```
+pip install wandb_workspaces
```
Comment (ngrayluna) [inline content/en/ref/python/wandb_workspaces/_index.md]:
```suggestion
pip install wandb-workspaces
```

File: content/en/ref/python/wandb_workspaces/_index.md [hunk 3]
```diff
@@ -1,17 +1,47 @@
...
 
+## Installation and setup
+
+### Sign up and create an API key
+
+To authenticate your machine with W&B, you must first generate an API key at https://wandb.ai/authorize.
+
+### Install and import packages
+
+Install the W&B library.
```
Comment (ngrayluna) [inline content/en/ref/python/wandb_workspaces/_index.md]:
```suggestion
Install the W&B Report and Workspaces library.
```

File: content/en/ref/python/wandb_workspaces/_index.md [hunk 4]
```diff
@@ -1,17 +1,47 @@
...
+
+```
+pip install wandb_workspaces
+```
+
+### Import W&B Python SDK:
+
+```python
+import wandb_workspaces
```
Comment (johndmulhausen) [inline content/en/ref/python/wandb_workspaces/_index.md]:
```suggestion
import wandb_workspaces.workspaces as ws
import wandb_workspaces.reports.v2 as wr 
```

File: content/en/ref/python/sdk/_index.md [hunk 5]
```diff
@@ -1,7 +1,35 @@
...
+```python
+import wandb
+```
+
+Specify the entity of your team in the following code block:
+
+```python
+TEAM_ENTITY = "<Team_Entity>" # Replace with your team entity
+PROJECT = "my-awesome-project"
+```
```
Comment (ngrayluna) [inline content/en/ref/python/sdk/_index.md]: Specifying an entity or project variable won't do much for the user. 

Let's add how to initialize a run instead? (since it's the atomic unit to W&B?) Maybe either:

```python
import wandb

# Specify your team entity
entity = "<team_entity>"

# Project that the run is recorded to
project = "my-awesome-project"

with wandb.init(entity=entity, project=project) as run:
   run.log({"accuracy": .90, "loss": .10})

```

Or something that actually generates some plots in their dashboard: 

```python
import wandb
import random

wandb.login()

# Specify your team entity
entity = "<team_entity>"

# Project that the run is recorded to
project = "my-awesome-project"

# Dictionary with hyperparameters
config = {
    'epochs' : 10,
    'lr' : 0.01
}

with wandb.init(entity=entity,project=project, config=config) as run:
    # simulating nose for demonstrative purposes
    offset = random.random() / 5

    # Simulate a training run
    for epoch in range(2, config['epochs']):
        acc = 1 - 2**-config['epochs'] - random.random() / config['epochs'] - offset
        loss = 2**-config['epochs'] + random.random() / config['epochs'] + offset
        print(f"epoch={config['epochs']}, accuracy={acc}, loss={loss}")
        run.log({"accuracy": acc, "loss": loss})
```

File: content/en/ref/python/sdk/functions/_index.md [hunk 6]
```diff
@@ -1,6 +1,59 @@
...
+        "learning_rate": 0.01,
+        "epochs": 10,
+    }
+)
+
+# Your training code here...
+
+# Finish the run
+wandb.finish()
+```
```
Comment (ngrayluna) [inline content/en/ref/python/sdk/functions/_index.md]: Heads up that we strongly advise not using wandb.finish(). Use context manager instead:

```suggestion
The most common workflow begins with authenticating with W&B, initializing a run, and logging values (such as accuracy and loss) from your training loop:


```python
import wandb

# Authenticate with W&B
wandb.login()

config = {
   "learning_rate": 0.01,
   "epochs": 10,
}

# Project where logs
project = "my-awesome-project"

# Start a new run
with wandb.init(project=project, config=config) as run:
   # Your training code here...
   
   # Log values to W&B
   run.log({"accuracy": acc, "loss": loss})
```

**Review by ngrayluna — CHANGES_REQUESTED** [review]

File: content/en/ref/python/sdk/_index.md (hunk 5 above)
Comment (johndmulhausen) [reply content/en/ref/python/sdk/_index.md]: Let's go with option 1 :)

**Comment by johndmulhausen** [comment]
Closing in favor of #1656 which was branched off of this work
